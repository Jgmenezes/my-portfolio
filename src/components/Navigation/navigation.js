import React from "react"
import * as S from "./navigation.styles"
import ROUTES from "../../../utils/constants/routes.constants";
import NavigationItem from "./NavigationItem/navigation-item";

const Navigation = () => (
     <S.HeaderWrapper>
          <S.Navigation>
            <S.List>
                {ROUTES.map(({
                    ACTIVE_BACKGROUND,
                    BACKGROUND,
                    IMAGE,
                    LABEL,
                    TO,
                  }) => (
                    <NavigationItem
                        activeBackground={ACTIVE_BACKGROUND}
                        background={BACKGROUND}
                        image={{ alt: '', src: IMAGE }}
                        title={LABEL}
                        to={TO}
                    />
                ))}
            </S.List>
          </S.Navigation>
     </S.HeaderWrapper>
);

export default Navigation;

