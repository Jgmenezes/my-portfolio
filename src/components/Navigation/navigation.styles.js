import styled, { keyframes } from "styled-components"
import MEDIA_QUERIES from "../../utils/constants/mediaQuerie.constants"

export const Label = styled.span`
  color: #ffffff;
  font-size: calc(0.6rem + 0.5vw);
  margin-top: 5px;
  z-index: 1;
`;

export const Item = styled.li`
  display: flex;
  flex: 100%;
  transition: all 0.5s ease;
`;

export const Navigation = styled.nav`
  display: flex;
  flex: 0 0 75%;
  
  @media ${MEDIA_QUERIES.DESKTOP_SCREEN} {
    height: 100%;
  }
`;

export const HeaderWrapper = styled.header`
  @media ${MEDIA_QUERIES.DESKTOP_SCREEN} {
    height: 100vh;
    left: 0;
    position: fixed;
    top: 0;
    width: 25vh;
  }
`;

export const List = styled.ul`
  display: flex;
  flex: 100%;
  list-style: none;
  margin: 0;
  padding: 0;
  
  @media ${MEDIA_QUERIES.DESKTOP_SCREEN} {
    flex-direction: column;
  }
`;

const kick = keyframes`
  0% {
    transform: scale(1.01);
  }
  
  50% {
    transform: scale(1.02);
  }
  
  100% {
    transform: scale(1.03);
  }
`;

export const Image = styled.img`
  height: 75px;
  width: 75px;
  z-index: 1;
  
  @media ${MEDIA_QUERIES.DESKTOP_SCREEN} {
    height: 45%;
    width: 100%;
    
    &:hover {
      animation: ${kick} 0.25s forwards;
    }
  }
`;