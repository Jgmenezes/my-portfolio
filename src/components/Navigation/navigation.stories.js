import React from 'react';
import { action } from '@storybook/addon-actions';
import Navigation from "./navigation";

export default {
    title: 'Components|Navigation',
    component: Navigation,
}

export const navigation = () => <Navigation />