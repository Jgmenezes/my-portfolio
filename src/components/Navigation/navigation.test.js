import React from "react"
import Navigation from "./navigation"
import { render } from "@testing-library/react"

describe('<Navigation />', () => {
    it('should render correctly', () => {
        const { asFragment } = render(<Navigation/>)
        expect(asFragment()).toMatchSnapshot()
    })
})