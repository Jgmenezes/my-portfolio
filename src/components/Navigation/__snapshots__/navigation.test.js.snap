// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render correctly 1`] = `
<DocumentFragment>
  <header
    class="sc-bxivhb bBDNGA"
  >
    <nav
      class="sc-htpNat oFzYK"
    >
      <ul
        class="sc-ifAKCX dsytzK"
      >
        <a
          activebackground="#5A58BF"
          aria-current="page"
          background="#312D4E"
          class="sc-bZQynM iKDMaF"
          href="/"
        >
          <img
            alt=""
            class="sc-gzVnrw hGhbNK"
            src="test-file-stub"
          />
          <span
            class="sc-htoDjs dIBGMA"
          >
            Home
          </span>
        </a>
        <a
          activebackground="#c14d7e"
          background="#242139"
          class="sc-bZQynM gjAAwc"
          href="/projects"
        >
          <img
            alt=""
            class="sc-gzVnrw hGhbNK"
            src="test-file-stub"
          />
          <span
            class="sc-htoDjs dIBGMA"
          >
            Projects
          </span>
        </a>
        <a
          activebackground="#4dc190"
          background="#1A1A29"
          class="sc-bZQynM fzgiDj"
          href="/about"
        >
          <img
            alt=""
            class="sc-gzVnrw hGhbNK"
            src="test-file-stub"
          />
          <span
            class="sc-htoDjs dIBGMA"
          >
            About
          </span>
        </a>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
